### PR TITLE
Add some more material to the introduction of step-18.

### LINKED ISSUE #1365: Update documentation of step-18
#212 changed step-18 to use the parallel::shared::Triangulation. Make sure that the introduction and results sections of this tutorial reflect this.

### PR DESCRIPTION
**Pull request by bangerth**
Also minor stylistic updates.

Fixes #1365.

### CODE REVIEW

File: examples/step-18/doc/intro.dox [hunk 1]
```diff
@@ -328,35 +329,84 @@ Both stress update and rotation are implemented in the function
...
+The code will generate the files <code>solution-TTTT.NNN.vtu</code>, 
+where <code>TTTT</code> is the
+timestep number (starting from 1) and <code>NNN</code> is the process rank
+(starting from
 0). These files contain the locally owned cells for the timestep and
-processor. The files solution-TTTT.visit is the visit record for timestep
-TTTT, while solution-TTTT.pvtu is the same for ParaView. Finally, the file
-solution.pvd is a special record only supported by ParaView that references
+processor. The files <code>solution-TTTT.visit</code> is the visit record for timestep
+<code>TTTT</code>, while <code>solution-TTTT.pvtu</code> is the same for ParaView. Finally, the file
```
Comment (davydden) [inline examples/step-18/doc/intro.dox]: visit can also open `.pvtu`. Actually i did not even know there is a special output for `Visit`.

File: examples/step-18/doc/intro.dox [hunk 2]
```diff
@@ -328,35 +329,84 @@ Both stress update and rotation are implemented in the function
...
+@endcode
+This knowledge extends to the DoFHandler object built on such triangulations,
+which can then identify which degrees of freedom are locally owned
+(see @ref GlossLocallyOwnedDofs) via calls such as 
+DoFHandler::n_locally_owned_dofs_per_processor() and
+DoFTools::extract_locally_relevant_dofs(). Finally, the DataOut class
+also knows how to deal with such triangulations and will simply skip
+generating graphical output on cells not locally owned.
+
+
```
Comment (davydden) [inline examples/step-18/doc/intro.dox]: maybe add a quick note about the limitation: triangulation is stored on each process; this eventually could become a bottleneck depending on the problem at hand; this is the core difference between this class and p::d::Tria (thus the names).

**Comment by davydden** [comment]
Apart from those minor comments, looks good to me :+1: Please, merge yourself.

File: examples/step-18/doc/intro.dox (hunk 1 above)
Comment (bangerth) [inline examples/step-18/doc/intro.dox]: Ha, interesting. That must be new since I didn't know that visit can read .pvtu files :-)

**Pull request merged**